syntax/expr_let_test.go

```diff
@@ -232,3 +217,33 @@ func TestExprLetGetPattern(t *testing.T) {
 
 	AssertCodeErrors(t, "", `let (x?: (k: (z?: w:42))) = (x: (y: (z: 1))); w`)
 }
+
+func TestExprLetDynIdentPattern(t *testing.T) {
+	AssertCodesEvalToSameValue(t, `42`, `let f = \x x * @{y}; let @{y} = 6; f(7)`)
+	AssertCodesEvalToSameValue(t, `42`, `let f = \x x * @{y}; let [@{y}, y] = [6, 8]; f(7)`)
+	AssertCodesEvalToSameValue(t, `42`, `let f = \x x * @{y}; let {@{y}, 8} = {6, 8}; f(7)`)
+	AssertCodesEvalToSameValue(t, `42`, `let f = \x x * @{y}; let (a:@{y}, ...) = (a:6, b:8); f(7)`)
+	AssertCodesEvalToSameValue(t, `42`, `let f = \x x * @{y}; let {1:@{y}, ...} = {1:6, 2:8}; f(7)`)
+	AssertCodesEvalToSameValue(t, `[2, 4, 8, 16, 32]`, `let f = \x x ^ @{y}; [1, 2, 3, 4, 5] >> \@{y} f(2)`)
+	AssertCodesEvalToSameValue(t, `[27, 64]`, `
```

Comment:
Nit: I don't know if it's just too boring or something, but these test cases would be easier to follow with simpler math, like 1 + @{y}.

Reply:
I made it more complicated, but I'm hoping the obvious pattern makes the intent clearer.